File: lib/limelight/players/radio_button.rb
#- Copyright 2008 8th Light, Inc.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

module Limelight
  module Players
    
    module RadioButton
      class << self
        def extended(prop)
          painter = Limelight::UI::Model::Painting::RadioButtonPainter.new(prop.panel)
          prop.radio_button = painter.radio_button
          prop.panel.painters << painter
          set_default_styles(prop)
        end
        
        def set_default_styles(prop)
          prop.style.width = "22"
          prop.style.height = "22"
        end
      end
  
      attr_accessor :radio_button
      attr_reader :button_group

      def group=(group_name)
        @button_group = scene.button_groups[group_name]
        @button_group.add(@radio_button)
      end
      
      def checked=(value)
        @radio_button.selected = value
      end
      alias :selected= :checked=
      
      def checked
        return @radio_button.is_selected
      end
      alias :checked? :checked
      alias :selected? :checked
      alias :selected :checked
      
    end
  
  end
end

File: src/limelight/ui/model/inputs/RadioButton.java
//- Copyright © 2008-2010 8th Light, Inc. All Rights Reserved.
//- Limelight and all included source files are distributed under terms of the GNU LGPL.

package limelight.ui.model.inputs;

import limelight.ui.RadioButtonGroup;
import limelight.ui.RadioButtonGroupMember;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionListener;
import java.awt.event.ActionEvent;

public class RadioButton extends JRadioButton implements RadioButtonGroupMember, ActionListener
{
  private final RadioButtonPanel panel;
  private RadioButtonGroup radioButtonGroup;

  public RadioButton(RadioButtonPanel panel)
  {
    this.panel = panel;
    this.addActionListener(this);
  }

  public void repaint()
  {
    if(panel != null && panel.getRoot() != null)
      panel.getRoot().addDirtyRegion(panel.getAbsoluteBounds());
  }

  public void repaint(long tm, int x, int y, int width, int height)
  {
    repaint(new Rectangle(x, y, width, height));
  }

  public void repaint(Rectangle dirtyBounds)
  {
    if(panel != null && panel.getRoot() != null)
    {
      Point location = panel.getAbsoluteLocation();
      dirtyBounds.translate(location.x, location.y);
      panel.getRoot().addDirtyRegion(dirtyBounds);
    }
  }

  public boolean isShowing()
  {
    return true;
  }

  public void setGroup(RadioButtonGroup radioButtonGroup)
  {
    this.radioButtonGroup = radioButtonGroup;
  }

  public void actionPerformed(ActionEvent e)
  {
    if(radioButtonGroup != null && isSelected())
      radioButtonGroup.buttonSelected(this);
  }
}
